### ISSUE TITLE
Configurable GoCQL Timeout Period

### ISSUE THREAD

**Issue by HadesArchitect** [HadesArchitect/GrafanaCassandraDatasource#88]
On the slow inefficient queries (for example, with `allow filtering`) we run into GoCQL timeout period.

Suggested solutions:
1. Just raise the timeout
2. Make timeout configurable

Possible improvement:
1. Warn a user about an inefficient query

**Comment by HadesArchitect** [comment]
>If you are using protocol version prior to 4, you can try to set Timeout paramter of cluster object in gocql to higher value. 
https://stackoverflow.com/questions/42922757/getting-timeout-error-with-gocql
Not sure why it relates to the protocol version

**Comment by HadesArchitect** [comment]
The important point, it can relate not to GoCQL settings, but to the C* cluster settings. If it is, we cannot do anything, only describe it in the "Known Issues" wiki page.

**Comment by HadesArchitect** [comment]
GoCQL settings example:
```
  var c *gocql.ClusterConfig
    if env.Production {
        c = gocql.NewCluster(
            "10.0.0.136",
            "10.0.0.137",
            "10.0.0.138",
            "10.0.0.139",
            "10.0.0.140",
        )
    } else {
        c = gocql.NewCluster("127.0.0.1")
    }
    c.Consistency = gocql.One
    c.Keyspace = CassandraKeyspace
    c.Timeout = 1 * time.Minute.               <-------- HERE
    c.NumConns = 1
    c.Compressor = gocql.SnappyCompressor{}
    return c.CreateSession()
    ```

**Comment by HadesArchitect** [comment]
To be released as 1.1.0
